aws_quota/check/vpc.py

```diff
@@ -295,3 +301,32 @@ def current(self) -> int:
             return len(list(filter(lambda cbas: cbas['Ipv6CidrBlockState']['State'] == 'associated', vpc['Ipv6CidrBlockAssociationSet'])))
         except KeyError:
             raise InstanceWithIdentifierNotFound(self)
+
+
+class VpcPeeringConnectionPerVpcCheck(InstanceQuotaCheck):
+    key = "vpc_peering_connection_per_vpc"
+    description = "Active VPC peering connections per VPC"
+    scope = QuotaScope.INSTANCE
+    service_code = 'vpc'
+    quota_code = 'L-7E9ECCDB'
+    instance_id = 'VPC ID'
+    used_services = ['ec2']
+
+    @staticmethod
+    def get_all_identifiers(session: boto3.Session) -> typing.List[str]:
+        return [vpc['VpcId'] for vpc in get_all_vpcs(get_client(session, 'ec2'))]
+
+    @property
+    def current(self) -> int:
+        try:
+            # Count unique VpcPeeringConnection if either requester or accepter VPC id matches.
+            result = set()
+            vpc_peerings = get_vpc_peering_connections(self.get_client('ec2'))
+            for vpc_peering in vpc_peerings:
+                if vpc_peering['Status']['Code'] == 'active' and (
+                        vpc_peering['AccepterVpcInfo']['VpcId'] == self.instance_id or
+                        vpc_peering['RequesterVpcInfo']['VpcId'] == self.instance_id):
+                    result.add(vpc_peering['VpcPeeringConnectionId'])
+            return len(result)
+        except Exception:
+            print("Error in VpcPeeringConnectionPerVpcCheck")
```

Comment:
I think we can just throw exceptions in checks e.g. raise new exception or completely remove "try" - the outer function will catch it and move on, in the future, do some nice failure reporting (for now it just prints exception message).

Reply:
Sg. I removed try-except. Agreed that we should define failure reporting so it is easy to tell what fails at which metric.

Also, we need to make sure the program should not fail on a single check, rather it should accumulate all errors and emit at the very end.

Reply:
Yup, one of the modifications in this branch already makes sure that check fail does not stop the process: https://github.com/jodzga/aws-quota-checker/blob/jarek-wip/aws_quota/cli.py#L107